tests/codegen/waiter-tests/src/main/kotlin/com/test/DefaultWaitersTestClient.kt

```diff
@@ -89,4 +89,34 @@ class DefaultWaitersTestClient<T>(resultList: List<Result<T>>) : WaitersTestClie
         @Suppress("UNCHECKED_CAST")
         return if (nextResult.isSuccess) (nextResult as Result<GetSubFieldProjectionResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
     }
+
+    override suspend fun getFunctionSumEquals(input: GetFunctionSumEqualsRequest): GetFunctionSumEqualsResponse {
+        val nextResult = results.next()
+        @Suppress("UNCHECKED_CAST")
+        return if (nextResult.isSuccess) (nextResult as Result<GetFunctionSumEqualsResponse>).getOrNull()!! else throw nextResult.exceptionOrNull()!!
```

Comment:
**Nit**: Seems like we could write a generic convenience method to collapse all this duplication in every operation.